extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/KafkaTestSupport.java

```diff
@@ -179,6 +243,33 @@ public <K, V> KafkaConsumer<K, V> createConsumer(
         return consumer;
     }
 
+    public void createSchemaRegistry(SchemaRegistryConfig config) throws Exception {
+        SchemaRegistryRestApplication schemaRegistryApplication = new SchemaRegistryRestApplication(config);
+        schemaRegistryServer = schemaRegistryApplication.createServer();
+        schemaRegistryServer.start();
+        schemaRegistry = schemaRegistryApplication.schemaRegistry();
+    }
+
+    public void shutdownSchemaRegistry() throws Exception {
+        if (schemaRegistryServer != null) {
+            schemaRegistryServer.stop();
+        }
+    }
+
+    public URI getSchemaRegistryURI() {
+        return schemaRegistryServer.getURI();
+    }
+
+    public int registerSchema(String subject, Schema schema) throws SchemaRegistryException {
+        return schemaRegistry.register(subject, new io.confluent.kafka.schemaregistry.client.rest.entities.Schema(
```

Comment:
you invoke directly the service, skipping REST API. It is a shortcut but for tests is should probably be fine.

Reply:
There is also `io.confluent.kafka.schemaregistry.client.SchemaRegistryClient`, which I later realized, but using `io.confluent.kafka.schemaregistry.storage.KafkaSchemaRegistry` directly is faster and more test-friendly (the client may not expose all functionality).